src/Access/ContextAccess.cpp

```diff
@@ -146,17 +146,23 @@ ContextAccess::ContextAccess(const AccessControl & access_control_, const Params
     : access_control(&access_control_)
     , params(params_)
 {
-    std::lock_guard lock{mutex};
+    
+}
 
-    subscription_for_user_change = access_control->subscribeForChanges(
-        *params.user_id, [this](const UUID &, const AccessEntityPtr & entity)
-    {
-        UserPtr changed_user = entity ? typeid_cast<UserPtr>(entity) : nullptr;
-        std::lock_guard lock2{mutex};
-        setUser(changed_user);
-    });
 
-    setUser(access_control->read<User>(*params.user_id));
+void ContextAccess::initialize()
+{
```

Comment:
```suggestion
{
    std::lock_guard lock{mutex};
```

Reply:
I think this lock is not necessary

Reply:
After we have subscribed for changes of `user` in `initialize()`, the lambda function can be called by another thread any moment. It can be called even before `setUser()` called in `initialize()` is returned. So without this lock there would be a small chance that `setUser()` will be called by two threads simultaneously.